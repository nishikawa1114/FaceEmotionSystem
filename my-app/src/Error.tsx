import React from 'react';
import './index.css';


interface errorProps {
    errorId: number;
    onSubmit: (e: React.ChangeEvent<HTMLFormElement>) => void;
}

export class Error extends React.Component<errorProps> {

    render() {

        let errorMessage: string;
        if (this.props.errorId == 1) {
            errorMessage = '画像が存在しません';
        } else {
            errorMessage = 'エラー'
        }

        return (
            <div>
                <header>
                    <h1>Face Emotion System</h1>
                </header>
                <div className="error_message">
                    <h2>{errorMessage}</h2>
                    <form onSubmit={this.props.onSubmit}>
                        <button type="submit">ホーム画面へ戻る</button>
                    </form>
                </div>
            </div>
        )
    }
}