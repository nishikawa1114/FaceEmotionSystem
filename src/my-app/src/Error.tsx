import React from 'react';
import './index.css';
import { ErrorId } from './types';

interface errorProps {
    errorId: number;
    onSubmit: (e: React.ChangeEvent<HTMLFormElement>) => void;
}

// エラー画面を表示するコンポーネント
export class Error extends React.Component<errorProps> {

    render() {

        let errorMessage: string;
        if (this.props.errorId === ErrorId.ERROR_IMAGE_NOT_EXIST) {
            errorMessage = '画像が存在しません';
        }else if (this.props.errorId === ErrorId.ERROR_ANALYZE_RESULT_EMPTY) {
            errorMessage = '分析結果が取得できませんでした';
        } else {
            errorMessage = '予期しないエラーが発生しました'
        }

        return (
            <div>
                <header>
                    <h1>Face Emotion System</h1>
                </header>
                <div className="error_message">
                    <h2>{errorMessage}</h2>
                    <form onSubmit={this.props.onSubmit}>
                        <button type="submit">ホーム画面へ戻る</button>
                    </form>
                </div>
            </div>
        )
    }
}