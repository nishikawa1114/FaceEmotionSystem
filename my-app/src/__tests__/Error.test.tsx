import {Error} from "../Error";
import { create } from "react-test-renderer";
import React from "react";

    it("snapshot test", () => {
        const tree = create(
            <Error
            errorId={0}
            onSubmit={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("error no image", () => {
        const tree2 = create(
            <Error
            errorId={1}
            onSubmit={() => {}}
            />
        ).toJSON();
        expect(tree2).toMatchSnapshot();
    })

    it("error analyze", () => {
        const tree3 = create(
            <Error
            errorId={2}
            onSubmit={() => {}}
            />
        ).toJSON();
        expect(tree3).toMatchSnapshot();
    })