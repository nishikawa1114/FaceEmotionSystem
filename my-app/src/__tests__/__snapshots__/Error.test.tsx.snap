// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error analyze 1`] = `
<div>
  <header>
    <h1>
      Face Emotion System
    </h1>
  </header>
  <div
    className="error_message"
  >
    <h2>
      分析結果が取得できませんでした
    </h2>
    <form
      onSubmit={[Function]}
    >
      <button
        type="submit"
      >
        ホーム画面へ戻る
      </button>
    </form>
  </div>
</div>
`;

exports[`error no image 1`] = `
<div>
  <header>
    <h1>
      Face Emotion System
    </h1>
  </header>
  <div
    className="error_message"
  >
    <h2>
      画像が存在しません
    </h2>
    <form
      onSubmit={[Function]}
    >
      <button
        type="submit"
      >
        ホーム画面へ戻る
      </button>
    </form>
  </div>
</div>
`;

exports[`snapshot test 1`] = `
<div>
  <header>
    <h1>
      Face Emotion System
    </h1>
  </header>
  <div
    className="error_message"
  >
    <h2>
      予期しないエラー
    </h2>
    <form
      onSubmit={[Function]}
    >
      <button
        type="submit"
      >
        ホーム画面へ戻る
      </button>
    </form>
  </div>
</div>
`;

exports[`snapshot test 2`] = `
<div>
  <header>
    <h1>
      Face Emotion System
    </h1>
  </header>
  <div
    className="error_message"
  >
    <h2>
      画像が存在しません
    </h2>
    <form
      onSubmit={[Function]}
    >
      <button
        type="submit"
      >
        ホーム画面へ戻る
      </button>
    </form>
  </div>
</div>
`;
